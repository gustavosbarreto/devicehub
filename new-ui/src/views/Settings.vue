<template>
  <v-card class="bg-v-theme-surface">
    <v-tabs background-color="secondary" stacked color="primary" align-tabs="center">
      <v-tab
        v-for="item in visibleItems"
        :key="item.title"
        :to="item.path"
        :data-test="item.title + '-tab'"
      >
        {{ item.title }}
      </v-tab>
    </v-tabs>

    <v-divider />
  </v-card>

  <v-card class="bg-v-theme-surface">
    <router-view />
  </v-card>
</template>

<script lang="ts">
import { computed, defineComponent } from "vue";
import { envVariables } from "../envVariables";
import { useStore } from "../store";

export default defineComponent({
  setup() {
    const store = useStore();

    const visibleItems = computed(() => items.value.filter((item) => !item.hidden));

    const currentInANamespace = computed(() => localStorage.getItem("tenant") !== "" );

    const hasNamespace = computed(() => store.getters["namespaces/getNumberNamespaces"] !== 0);

    const items = computed(() => [
      {
        title: "Profile",
        path: "/settings",
      },
      {
        title: "Namespace",
        path: "/settings/namespace-manager",
        hidden: !currentInANamespace.value,
      },
      {
        title: "Private Keys",
        path: "/settings/private-keys",
      },
      {
        title: "Tags",
        path: "/settings/tags",
      },
      {
        title: "Billing",
        path: "/settings/billing",
        hidden: !(
          envVariables.billingEnable &&
          envVariables.isCloud &&
          hasNamespace.value
        ),
      },
    ]);

    return {
      items,
      visibleItems,
    };
  },
});
</script>
