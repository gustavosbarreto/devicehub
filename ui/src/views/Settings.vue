<template>
  <v-card class="bg-v-theme-surface" data-test="settings-card">
    <v-tabs background-color="secondary" stacked color="primary" align-tabs="center">
      <v-tab
        v-for="item in visibleItems"
        :key="item.title"
        :to="item.path"
        :data-test="item.title + '-tab'"
      >
        {{ item.title }}
      </v-tab>
    </v-tabs>

    <v-divider />
  </v-card>

  <v-card class="bg-v-theme-surface">
    <router-view />
  </v-card>
</template>

<script setup lang="ts">
import { computed } from "vue";
import { envVariables } from "../envVariables";
import { useStore } from "../store";

const store = useStore();

const currentInANamespace = computed(() => localStorage.getItem("tenant") !== "");

const hasNamespace = computed(() => store.getters["namespaces/getNumberNamespaces"] !== 0);

const items = computed(() => [
  {
    title: "Profile",
    path: "/settings",
  },
  {
    title: "Namespace",
    path: "/settings/namespace",
    hidden: !currentInANamespace.value,
  },
  {
    title: "Private Keys",
    path: "/settings/private-keys",
  },
  {
    title: "Tags",
    path: "/settings/tags",
  },
  {
    title: "Billing",
    path: "/settings/billing",
    hidden: !(
      envVariables.billingEnable
          && envVariables.isCloud
          && hasNamespace.value
    ),
  },
]);

const visibleItems = computed(() => items.value.filter((item) => !item.hidden));

defineExpose({ visibleItems });
</script>
