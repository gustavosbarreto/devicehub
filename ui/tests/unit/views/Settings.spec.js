import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Settings from '@/views/Settings';

describe('Settings', () => {
  const localVue = createLocalVue();
  localVue.use(VueRouter);
  localVue.use(Vuex);

  let wrapper;

  const items = [
    {
      title: 'Profile',
      path: '/settings',
    },
    {
      title: 'Namespace',
      path: '/settings/namespace-manager',
    },
    {
      title: 'Private Keys',
      path: '/settings/private-keys',
    },
  ];

  beforeEach(() => {
    wrapper = shallowMount(Settings, {
      localVue,
      stubs: ['fragment'],
      mocks: {
        $env: (isEnterprise) => isEnterprise,
      },
    });
  });

  ///////
  // Component Rendering
  //////

  it('Is a Vue instance', () => {
    expect(wrapper).toBeTruthy();
  });
  it('Renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  ///////
  // Data and Props checking
  //////

  it('Compare data with default value', () => {
    expect(wrapper.vm.drawer).toEqual(true);
    expect(wrapper.vm.clipped).toEqual(false);
  });

  //////
  // HTML validation
  //////

  it('Renders the template with data', async () => {
    Object.keys(items).forEach((item) => {
      expect(wrapper.find(`[data-test="${items[item].title}-tab"]`).text()).toEqual(items[item].title);
    });
  });
});
