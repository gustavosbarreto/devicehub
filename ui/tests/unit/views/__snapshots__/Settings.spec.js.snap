// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Renders the component 1`] = `
<v-container-stub tag="div">
  <v-card-stub loaderheight="4" tag="div" elevation="0">
    <v-app-bar-stub color="transparent" tag="header" extensionheight="48" flat="true" src="" value="true">
      <v-tabs-stub activeclass="" centered="true" nexticon="$next" previcon="$prev" slidersize="2">
        <v-tab-stub to="/settings" ripple="true">
          Profile
        </v-tab-stub>
        <v-tab-stub to="/settings/namespace-manager" ripple="true">
          Namespace
        </v-tab-stub>
        <v-tab-stub to="/settings/private-keys" ripple="true">
          Private Keys
        </v-tab-stub>
      </v-tabs-stub>
    </v-app-bar-stub>
    <v-divider-stub></v-divider-stub>
    <v-container-stub tag="div" fluid="true" class="pa-0">
      <router-view-stub name="default"></router-view-stub>
    </v-container-stub>
  </v-card-stub>
</v-container-stub>
`;
